### PR TITLE
Create an issue if an on push gradle check fails.

### PR DESCRIPTION
**Pull request by mch2**
### Description
Creates an issue if a gradle check run triggered by a push fails.  Tested this out from a test repo [here](https://github.com/mch2/TestActions/blob/main/.github/workflows/test.yml).

### Issues Resolved
closes https://github.com/opensearch-project/OpenSearch/issues/5799

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/failed_check.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+title: Gradle Check Failure.
+labels: >test-failure bug
+---
+
+A gradle check workflow has failed after merge.
+
+PR: {{ env.workflow_url }}
+CommitId: {{ env.pr_from_sha }}
```
Comment (reta) [inline .github/ISSUE_TEMPLATE/failed_check.md:9]: Just an idea, should we tag the author here (if possible) so he/she gets notified about post merge failure?

Comment (mch2) [reply .github/ISSUE_TEMPLATE/failed_check.md:9]: Added both the pusher and PR author as assignees.

**Review by reta — APPROVED** [review]

**Comment by andrross** [comment]
@mch2 Are we going to backport this so that issues get created for failures on the 2.x branch as well?

**Comment by mch2** [comment]
> @mch2 Are we going to backport this so that issues get created for failures on the 2.x branch as well?

Just added backport label, I originally thought main was enough but we also get separate check runs on 2.x post merge.

**Pull request merged**